test/snippet/search/bi_fm_index_cursor.cpp

```diff
@@ -13,106 +13,106 @@ int main()
 debug_stream << "Example extend_left(seq)\n";
 //! [extend_left_seq]
 std::vector<dna4> genome{"GAATTAATGAAC"_dna4};
-bi_fm_index index{genome}; // build the index
-
-auto it = index.begin();         // create an cursor
-it.extend_right("AAC"_dna4);     // search the sequence "AAC"
-debug_stream << it.query() << '\n'; // outputs "AAC"
-it.extend_left("ATG"_dna4);      // extend the query to "ATGAAT"
-                                 // The rightmost character of "ATG" is extended to the left first.
-debug_stream << it.query() << '\n'; // outputs "ATGAAT"
+bi_fm_index<false> index{genome};          // build the bidirectional index
+
+auto cur = index.begin();                  // create a cursor
+cur.extend_right("AAC"_dna4);              // search the sequence "AAC"
+debug_stream << cur.query(genome) << '\n'; // outputs "AAC"
+cur.extend_left("ATG"_dna4);               // extend the query to "ATGAAT"
+                                           // The rightmost character of "ATG" is extended to the left first.
+debug_stream << cur.query(genome) << '\n'; // outputs "ATGAAT"
```

Comment:
I don't get this "feature". Why do I want to print my query again?

Reply:
If you work with the iterators and want to know where you actually went, i.e., where you went in the suffix tree. That reminds me, we wanted to rename it to `path_label`.